resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/KubernetesClientBuilder.scala

```diff
@@ -25,29 +25,60 @@ import io.fabric8.kubernetes.client.{Config, ConfigBuilder, DefaultKubernetesCli
 import org.apache.spark.deploy.kubernetes.constants._
 
 private[spark] object KubernetesClientBuilder {
-  private val API_SERVER_TOKEN = new File(Config.KUBERNETES_SERVICE_ACCOUNT_TOKEN_PATH)
-  private val CA_CERT_FILE = new File(Config.KUBERNETES_SERVICE_ACCOUNT_CA_CRT_PATH)
+  private val SERVICE_ACCOUNT_TOKEN = new File(Config.KUBERNETES_SERVICE_ACCOUNT_TOKEN_PATH)
+  private val SERVICE_ACCOUNT_CA_CERT = new File(Config.KUBERNETES_SERVICE_ACCOUNT_CA_CRT_PATH)
+  private val MOUNTED_CREDENTIALS_BASE_DIR = new File(
+    DRIVER_CONTAINER_KUBERNETES_CREDENTIALS_SECRETS_BASE_DIR)
+  private val MOUNTED_TOKEN = new File(MOUNTED_CREDENTIALS_BASE_DIR,
+    DRIVER_CONTAINER_OAUTH_TOKEN_SECRET_NAME)
+  private val MOUNTED_CA_CERT = new File(MOUNTED_CREDENTIALS_BASE_DIR,
+    DRIVER_CONTAINER_CA_CERT_FILE_SECRET_NAME)
+  private val MOUNTED_CLIENT_KEY = new File(MOUNTED_CREDENTIALS_BASE_DIR,
+    DRIVER_CONTAINER_CLIENT_KEY_FILE_SECRET_NAME)
+  private val MOUNTED_CLIENT_CERT = new File(MOUNTED_CREDENTIALS_BASE_DIR,
+    DRIVER_CONTAINER_CLIENT_CERT_FILE_SECRET_NAME)
 
   /**
    * Creates a {@link KubernetesClient}, expecting to be from
    * within the context of a pod. When doing so, credentials files
    * are picked up from canonical locations, as they are injected
    * into the pod's disk space.
    */
-  def buildFromWithinPod(
-      kubernetesNamespace: String): DefaultKubernetesClient = {
+  def buildFromWithinPod(kubernetesNamespace: String): DefaultKubernetesClient = {
     var clientConfigBuilder = new ConfigBuilder()
       .withApiVersion("v1")
       .withMasterUrl(KUBERNETES_MASTER_INTERNAL_URL)
       .withNamespace(kubernetesNamespace)
 
```

Comment:
The service account token file is actually always present, so this check wouldn't be correct.